### DISCUSSION TITLE
neorg-roam

### DISCUSSION THREAD

**Discussion by Jarvismkennedy** [nvim-neorg/neorg#1078] [Ideas]
I used emacs for a (very) short time, specifically for org mode and being able to quickly take notes while working. I didn't get a ton of experience with emacs and org-mode but one of the features I found very useful was org-roam.

I eventually came back to neovim and started using neorg. I am not sure how much of this can already be accomplished with the core modules, but I decided to start working on a neorg-roam module. Its the first neovim plugin I have worked on and definitely a work in progress. Its already useful to me so I thought I would share it here in case anyone is looking for something similar. Here is the repo: https://github.com/Jarvismkennedy/neorg-roam.nvim

The currently implemented features are:

- Search the current workspace to open / create new notes.
- Insert links to other norg files in the workspace.
- Capture to existing or new notes with capture templates.

Currently on the to-do list:

- Create a sqlite module and implement back links.
- Insert links is broken when inserting into a norg file not in the current workspace.
- Move files / rename files breaks links.
- Improvements to the capture templates.

**Comment by max397574** [comment]
neorg-telescope already has functionality to insert links btw

**Reply by Jarvismkennedy** [reply]
That's good to know, I will check out neorg-telescope so that I am not re writing modules that already exist.
